modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/near/GridNearTxLocal.java

```diff
@@ -2851,6 +2868,74 @@ public void addKeyMapping(IgniteTxKey key, ClusterNode node) {
     }
 
     /**
+     * Suspends transaction. It could be resumed later. Supported only for optimistic transactions.
+     *
+     * @throws IgniteCheckedException If the transaction is in an incorrect state, or timed out.
+     */
+    public void suspend() throws IgniteCheckedException {
+        if (log.isDebugEnabled())
+            log.debug("Suspend near local tx: " + this);
+
+        if (pessimistic() || system()) {
```

Comment:
suspendInProgress unlocks change of threadId outside instance.
See threadId(long threadId) method.